### PR TITLE
Initial version

### PR DESCRIPTION
**Pull request by doudou**
Depends on:
- [x] https://github.com/rock-drivers/drivers-orogen-canopen_master/pull/1
- [x] https://github.com/tidewise/drivers-motors_roboteq_canopen/pull/2

### CODE REVIEW

**Comment by doudou** [comment]
Ping @g-arjones

File: tasks/Task.cpp [hunk 1]
```diff
@@ -13,37 +17,214 @@ Task::~Task()
 {
 }
 
+bool Task::configureHook()
+{
+    delete m_driver;
+    delete m_state_machine;
```
Comment (g-arjones) [inline tasks/Task.cpp]: I can't find the reference in the standard by I don't think you can expect these to be null-initialized

File: tasks/Task.cpp [hunk 2]
```diff
@@ -13,37 +17,214 @@ Task::~Task()
...
     TaskBase::stopHook();
 }
 void Task::cleanupHook()
 {
+    channelsToSwitchOnDisabled();
     TaskBase::cleanupHook();
 }
+
+void Task::waitDS402State(Channel& channel, StatusWord::State state) {
+    while (true) {
```
Comment (g-arjones) [inline tasks/Task.cpp:158]: No timeout?

File: tasks/Task.cpp (hunk 1 above)
Comment (doudou) [reply tasks/Task.cpp]: Indeed not, but they were initialized in the attribute declaration. However, they were double-delete codepaths possible. I'm now setting them to nullptr after the delete.

File: tasks/Task.cpp (hunk 2 above)
Comment (doudou) [reply tasks/Task.cpp:158]: Ouch. Good catch.

**Review by g-arjones — APPROVED** [review]

**Pull request merged**